### PR TITLE
Exclude the vendor as including vendor will cause CI break after merge #48287

### PR DESCRIPTION
**Pull request by Eric-Guo**
### Motivation / Background

This Pull Request has been created because #48287 break some CI in [my](https://git.thape.com.cn/Eric-Guo/coreui-pro-rails-starter/-/jobs/103123#L565) and other pipeline.

@rafaelfranca suggest [exclude the vendor folder](https://github.com/rails/rails/pull/48287#issuecomment-1563170637), I tested and it works.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [ ] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: railties/lib/rails/engine.rb [hunk 1]
```diff
@@ -619,7 +619,7 @@ def load_seed
       next if is_a?(Rails::Application)
 
       fixtures = config.root.join("test", "fixtures")
-      if fixtures.exist? && fixtures.to_s.start_with?(Rails.root.to_s)
+      if fixtures.exist? && fixtures.to_s.start_with?(Rails.root.to_s) && !fixtures.to_s.start_with?(Rails.root.join("vendor").to_s)
```
Comment (zzak) [inline railties/lib/rails/engine.rb:622]: Sorry I don't have a better alternative off-hand, but I wonder if 3 conditionals on a single line is a good reason to pull this into a separate method?

Comment (Eric-Guo) [reply railties/lib/rails/engine.rb]: @zzak Changed, the testing failure is [not relative](https://buildkite.com/rails/rails/builds/96731#018865e7-ef11-4463-b0ab-cda513be9d7f/1102-1123) with this PR. (https://saucelabs.com/rest/v1/rubyonrails/tunnels - 500 (Internal Server Error) Error)

Comment (zzak) [reply railties/lib/rails/engine.rb]: @Eric-Guo Nice! Maybe "fixtures_in..." plural?

Comment (Eric-Guo) [reply railties/lib/rails/engine.rb]: Changed.

**Comment by shioyama** [comment]
Thanks for this! Sorry I honestly hadn't expected https://github.com/rails/rails/pull/48287 to be merged so quickly. I'm glad I at least excluded everything outside of root, but hadn't even thought of `"vendor"`.

**Review by zzak — APPROVED** [review]

**Pull request merged**